core/executor/src/precompiles/modexp.rs

```diff
@@ -102,7 +102,7 @@ fn get_data(data: &[u8], mut start: usize, size: usize) -> Result<Integer, Preco
             exit_status: ExitError::StackOverflow,
         })?;
 
-    padded.extend(std::iter::repeat(0).take(size - (end - start)));
+    padded.extend(std::iter::repeat(0).take(size - (end.saturating_sub(start))));
```

Comment:
How about using `padded.resize(size, 0)`?

Reply:
not suitable, slower

`resize` use this thing internally：
https://doc.rust-lang.org/nightly/src/alloc/vec/mod.rs.html#2476-2498

`extend` use this:
https://doc.rust-lang.org/nightly/src/alloc/vec/spec_extend.rs.html#49-57
https://doc.rust-lang.org/nightly/src/alloc/vec/mod.rs.html#1939-1945

Reply:
Why would the specialized version be used? The iterator here is not `std::vec::IntoIter`. And I doubt there's much difference after optimization.

Reply:
I see similar output for

```rust
pub fn f(v: &mut Vec<u8>, len: usize) {
    v.extend(std::iter::repeat(0).take(len.saturating_sub(v.len())));
}

pub fn g(v: &mut Vec<u8>, len: usize) {
    v.resize(len, 0);
}
```

Both will reserve and `memset`.

Reply:
Just FYI, `extend` is a tiny bit more efficient than `resize` because it doesn't need to handle shrinking. On setting both will use `memset`.